var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -0,0 +1,117 @@
+from spack import *
+import sys, os, re
+
+def filter_pick(input_list, regex_filter):
+    return [l for l in input_list for m in (regex_filter(l),) if m]
+
+def unfilter_pick(input_list, regex_filter):
+    return [l for l in input_list for m in (regex_filter(l),) if not m]
+
+def get_all_components():
+    all_components = []
+    with open("pset/mediaconfig.xml", "r") as f:
+        lines = f.readlines()
+        for line in lines:
+            if line.find('<Abbr>') != -1:
+                component = line[line.find('<Abbr>') + 6:line.find('</Abbr>')]
+                all_components.append(component)
+    return all_components
```

Comment:
yeah, silly me, I even already defined a get_all_components function in intel/package.py, just didn't use it. I will add a doc string for that function too.